debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectorIT.java

```diff
@@ -256,6 +257,66 @@ public void streamChangesWhileStopped() throws Exception {
         }
     }
 
+    @Test
+    @FixFor("DBZ-1069")
+    public void verifyOffsets() throws Exception {
+        final int RECORDS_PER_TABLE = 5;
+        final int TABLES = 2;
+        final int ID_START = 10;
+        final int ID_RESTART = 100;
+        final Configuration config = TestHelper.defaultConfig()
+                .with(SqlServerConnectorConfig.SNAPSHOT_MODE, SnapshotMode.INITIAL)
+                .build();
+
+        for (int i = 0; i < RECORDS_PER_TABLE; i++) {
+            final int id = ID_START + i;
+            connection.execute(
+                    "INSERT INTO tablea VALUES(" + id + ", 'a')"
+            );
+            connection.execute(
+                    "INSERT INTO tableb VALUES(" + id + ", 'b')"
+            );
+        }
+
+        start(SqlServerConnector.class, config);
+        assertConnectorIsRunning();
+
+        List<SourceRecord> records = consumeRecordsByTopic(1 + RECORDS_PER_TABLE * TABLES).allRecordsInOrder();
+        records = records.subList(1, records.size());
+        for (Iterator<SourceRecord> it = records.iterator(); it.hasNext();) {
+            SourceRecord record = it.next();
+            assertThat(record.sourceOffset().get("snapshot")).as("Snapshot phase").isEqualTo(true);
+            if (it.hasNext()) {
+                assertThat(record.sourceOffset().get("snapshot_completed")).as("Snapshot in progress").isEqualTo(false);
+            }
+            else {
+                assertThat(record.sourceOffset().get("snapshot_completed")).as("Snapshot completed").isEqualTo(true);
+            }
+        }
+
+        stopConnector();
+        for (int i = 0; i < RECORDS_PER_TABLE; i++) {
+            final int id = ID_RESTART + i;
+            connection.execute(
+                    "INSERT INTO tablea VALUES(" + id + ", 'a')"
+            );
+            connection.execute(
+                    "INSERT INTO tableb VALUES(" + id + ", 'b')"
+            );
+        }
+
+        start(SqlServerConnector.class, config);
+        assertConnectorIsRunning();
+        records = consumeRecordsByTopic(1 + RECORDS_PER_TABLE * TABLES).allRecordsInOrder();
+        records = records.subList(1, records.size());
```

Comment:
It seems to be wrong. If you have a look at records, they have ids: 10, 11, 12, ... Should be 100, 101.

Reply:
This means that the same records have been fetched twice - during initial load and streaming.

Reply:
You are right, but this is a different issue - I need to think about it more. The last snapshot record should probably contan the LSN from which we should start reading in streaming phase.